plugins/module_utils/network/sonic/facts/snmp/snmp.py

```diff
@@ -0,0 +1,406 @@
+        agentaddress_config = snmp_list['engine']['listen']
+
+        for agentaddress in agentaddress_config:
+            agentaddress_dict['interface'] = agentaddress.get('udp').get("ietf-snmp-ext:interface")
+            agentaddress_dict['ip'] = agentaddress.get('udp').get("ip")
+            agentaddress_dict['port'] = agentaddress.get('udp').get("port")
+            autogen_agentaddress.append({'agent-entry-name': agentaddress.get('name'), 'address': {'ip': agentaddress_dict['ip']}})
+
+            agentaddress_list.append(agentaddress_dict)
+        return agentaddress_list
+
+    def get_snmp_community(self, snmp_list):
+        """
+        Get snmp community from the snmp list
+        """
+        community_list = []
+
+        if not snmp_list.get('community'):
+            return community_list
+        community_config = snmp_list['community']
+
+        for community in community_config:
+            if community is None:
+                return community_list
+            community_list.append({"name": community.get("index"), "group": community.get("security-name")})
+
+        return community_list
+
+    def get_snmp_engine(self, snmp_list):
+        """
+        Get snmp engine from the snmp list
+        """
+        engine = ''
+
+        if not snmp_list.get('engine'):
+            return engine
+
+        engine_config = snmp_list['engine'].get('engine-id')
+
+        return engine_config
+
+    def get_snmp_users(self, snmp_list):
+        """
+        Get snmp users from the snmp list
+        """
+        user_list = []
+        group_config = []
+
+        if not snmp_list.get('usm') or not snmp_list.get('usm').get('local') or not snmp_list.get('usm').get('local').get('user'):
+            return user_list
+
+        if snmp_list.get('vacm') and snmp_list.get('vacm').get('group'):
+            group_config = snmp_list['vacm']['group']
+
+        user_config = snmp_list['usm']['local'].get('user')
+
+        for user in user_config:
+            user_dict = {}
+            auth_type = list(user.get('auth').keys())[0]
+            auth_key = user.get("auth").get(auth_type).get('key')
+
+            priv_type = list(user.get('priv').keys())[0]
+            priv_key = user.get("priv").get(priv_type).get('key')
+
+            matched_user = None
+            group_name = ""
+            for group in group_config:
+                if group.get('member') is not None:
+                    matched_user = next((member for member in group['member'] if member['security-name'] == user['name']), None)
+                    if matched_user is not None:
+                        group_name = group['name']
+                        break
+            user_dict['group'] = group_name
+            user_dict['name'] = user.get('name')
+            user_dict['auth'] = {'auth_type': auth_type, 'key': auth_key}
+            user_dict['priv'] = {'priv_type': priv_type, 'key': priv_key}
+            user_dict['encrypted'] = user.get('encrypted')
```

Comment:
Please retrieve encrypted status. With this change, the encrypted option in user input (want) need not be popped in config.
```suggestion
            user_dict['encrypted'] = user.get('ietf-snmp-ext:encrypted')
```